go/tasks/plugins/k8s/sagemaker/sagemaker.go

```diff
@@ -43,7 +44,7 @@ type awsSagemakerPlugin struct {
 }
 
 func (m awsSagemakerPlugin) BuildIdentityResource(_ context.Context, _ pluginsCore.TaskExecutionMetadata) (k8s.Resource, error) {
-	if m.TaskType == trainingJobTaskType {
+	if m.TaskType == trainingJobTaskType || m.TaskType == customTrainingJobTaskType {
```

Comment:
IMO, This file should be split into 3/4 files
1. builtin.go
2. custom.go
3. hpo.go
4. status.go / plugin.go
Thus the flow would be go to plugin.go, if custom then jump to custom.go and so on
all the status is in plugin.go
All resource building is in individual files.

NOTE: In Golang, the same struct methods can be spread over multiple files. This optimizes readability and I prefer that.

Reply:
yup I agree.